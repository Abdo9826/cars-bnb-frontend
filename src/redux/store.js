import { configureStore } from '@reduxjs/toolkit';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import logger from 'redux-logger';
import CarsReducer from './Cars/carSlice';
import authenticationReducer from './Auth/authenticationSlice';
import ReserveReducer from './Reservations/ReserveSlice';
import reservationListReducer from './reservationList/reservationListSlice';

export default configureStore({
  reducer: {
    user: authenticationReducer,
    cars: CarsReducer,
    addReservations: ReserveReducer,
    reservations: reservationListReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});
