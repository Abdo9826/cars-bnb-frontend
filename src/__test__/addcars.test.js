import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Addcars from '../components/cars/addcars';
import store from '../redux/store';

it('To check test is working', () => {
  expect(true).toBe(true);
});

it('Test if header render correctly', () => {
  const TREE = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Addcars />
      </BrowserRouter>
    </Provider>,
  );
  expect(TREE).toMatchSnapshot();
});
