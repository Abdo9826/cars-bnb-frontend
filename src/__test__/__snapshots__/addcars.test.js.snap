// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if header render correctly 1`] = `
<div
  className="big"
>
  <div
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="brand-logo"
      >
         
        <img
          alt="Cars bnb logo"
          src="cars-bnb-logo.png"
        />
         
      </div>
      <div
        className="inputs"
      >
        <label
          className="green"
        >
          Model
        </label>
        <input
          className="input2"
          name="model"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Model"
          type="text"
        />
        <label
          className="green"
        >
          Brand
        </label>
        <input
          className="input2"
          name="brand"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Brand"
          type="text"
        />
        <label
          className="green"
          htmlFor="date"
        >
          date
        </label>
        <input
          className="input2"
          id="date"
          max="2022-12-31"
          min="1990-01-01"
          name="year_released"
          onBlur={[Function]}
          onChange={[Function]}
          type="date"
        />
        <label
          className="green"
        >
          car Photo
        </label>
        <input
          className="input2"
          name="featured_image"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="car Photo"
          type="text"
        />
        <label
          className="green"
        >
          specifications
        </label>
        <textarea
          className="input3"
          name="specifications"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="specifications"
          type="text"
        />
        <button
          className=" mr"
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</div>
`;
